Handle reverse deps on global variables in libs (#3161)

leamas opened the pull request:
Remove all references to OpenCPN core global variables in libs.

Some has been discovered being unused and are thus just remoged. Other are replaces with callbacks or parameters.

If/when Håkan has time he might do some testing in opencpn-libs. 


Closes: #3161

bdbcat commented:
This is a big push.
OK if I bring them in a few at a time, giving more granularity?

leamas commented:
> This is a big push.

But not **that**  big, here was a lot of cruft which was made part of it by a late mistake. Rebased, removed this garbage..

> OK if I bring them in a few at a time, giving more granularity?

Of course. Furthermore, this is about #3161 so it probably makes sense to push the 5.8.2 tag before merging it at all. 

As for 5.8.2, #3139 is the important one.

leamas commented:
Rebased once again to make it easier to merge in parts.  The natural points would be

  1. c05589888 `ChartCtxFactory: refactor duplicated code.`:  All changes to ss52plib
  2. e608b9df0 nmea0183: Re-introduce default ctor for use in plugins.: All changes to nmea0183
  3. eb58c77df  texcmp: Remove unused "extern g_throttle_squish"  (3161): Other changes.
